Source_Code/Server/CardDeck.java

```diff
@@ -0,0 +1,143 @@
+    * cards that are not assigned to the players.
+    */
+   public static SolutionHand shuffleAndAssignCards(
+      ArrayList< Player > players )
+   {
+      int          counter       = 0; // Iterator pointer to keep track of the
+                                      // cards
+      SolutionHand solutionCards = null;
+
+      /*
+       * Grab all the names and types of each card Had to do some list
+       * manipulation to get the enums into a modifiable list. Strings may be
+       * easier to handle, but that depends on how we implement the
+       * customization feature later. TODO: Explore using strings here
+       * instead.
+       */
+      List< WeaponType >    weaponEnums    = new ArrayList< WeaponType >(
+         Arrays.asList( WeaponCard.WeaponType.values() ) );
+      List< CharacterName > characterEnums = new ArrayList< CharacterName >(
+         Arrays.asList( CharacterCard.CharacterName.values() ) );
+      List< RoomName >      roomEnums      = new ArrayList< RoomName >(
+         Arrays.asList( RoomCard.RoomName.values() ) );
+
+      System.out.println( characterEnums.size() );
+
+      // Card equivalents of the enums
+      ArrayList< CharacterCard > characters = new ArrayList< CharacterCard >();
+      ArrayList< RoomCard > rooms = new ArrayList< RoomCard >() ;
+      ArrayList< WeaponCard > weapons = new ArrayList< WeaponCard >();
+
+      for( CharacterName c : characterEnums )
+      {
+         characters.add( new CharacterCard( c ) );
+      }
+
+      for( RoomName r : roomEnums )
+      {
+         rooms.add( new RoomCard( r ) );
+      }
+
+      for( WeaponType w : weaponEnums )
+      {
+         weapons.add( new WeaponCard( w ) );
+      }
+
+      // Shuffle the ArrayLists
+      Collections.shuffle( weapons );
+      Collections.shuffle( characters );
+      Collections.shuffle( rooms );
+
+      System.out.println( characters.get( 0 ).getCharacterName() );
+
+      // Remove the top card from each card type and store as solution cards
+      try
+      {
+         solutionCards = new SolutionHand( characters.remove( 0 ),
+            rooms.remove( 0 ), weapons.remove( 0 ) );
+         
+      } catch( Exception e )
+      {
+         System.out.println( "Error generating solution cards." );
+         e.printStackTrace();
+         System.exit( 0 );
+      }
+
+      // Combine all cards into one larger deck
+      List< Card > deck = new ArrayList< Card >();
+      deck.addAll( characters );
+
+      deck.addAll( rooms );
+
+      deck.addAll( weapons );
+      
+
+      // Shuffle the deck and assign to each player sequentially
+      Collections.shuffle( deck );
+
+
+      /*
+       * Loops through the cards and players, assigning the cards to players
+       * one by one.
+       */
+      for( int i = 0; i < deck.size() - 1; i++ )
+      {
+
+         // Resets the player counter back to the start of the list.
+         if( counter > players.size() - 1 )

```

Comment:
I think counter >= size is a bit easier to follow and saves us an operation.

Could also remove the if and just do "players.get( counter % players.size()).setCard( deck.get( i ) );", although I don't actually know if that's more efficient or if it just appeals to my desire to write one-liners